### PR TITLE
[PATCH v7] validation: ipsec: reduce code duplication in reassembly test cases

### PR DESCRIPTION
**Pull request by JannePeltonen**
This PR depends on PR 992 (api: reassembly) and adds one commit on top of it. The commit reduces code duplication and makes reassembly test case code simpler as suggested in the review of PR 992. @anoobj please review.

v2: Rebased on top of latest master that now includes PR 992 commits.

### CODE REVIEW

**Comment by JannePeltonen** [comment]
v3: rebased

File: test/validation/api/ipsec/ipsec_test_in.c [hunk 1]
```diff
@@ -1764,132 +1767,108 @@ static void test_ipsec_sa_print(void)
...
+
+		/*
+		 * Perform inbound IPsec processing for the IPsec packet.
+		 * Expect result packet only for the last packet.
+		 */
+		memset(&test_in, 0, sizeof(test_in));
+		if (i == num_input_packets - 1) {
+			part_prep_plain(&test_in, 1, is_result_ipv6, true);
+			test_in.out[0].pkt_res = result_packet;
+		}		ipsec_test_packet_from_pkt(&test_pkt, &pkt);
```
Comment (JereLeppanen) [inline test/validation/api/ipsec/ipsec_test_in.c:1803]: Missing newline?

Comment (JannePeltonen) [reply test/validation/api/ipsec/ipsec_test_in.c:1803]: Yes. No idea who ate it. Will fix in the next version.

Comment (JannePeltonen) [reply test/validation/api/ipsec/ipsec_test_in.c]: Fixed in v4.

**Comment by JannePeltonen** [comment]
v4: rebased and added the missing newline.

**Review by JereLeppanen — APPROVED** [review]

**Comment by JannePeltonen** [comment]
v5: Added review tag.

File: test/validation/api/ipsec/ipsec_test_in.c [hunk 2]
```diff
@@ -1764,132 +1767,109 @@ static void test_ipsec_sa_print(void)
...
 
 static void test_in_ipv4_esp_reass_success_four_frags_ooo(odp_ipsec_sa_t out_sa,
 							  odp_ipsec_sa_t in_sa)
 {
-	ipsec_test_part test_out[MAX_FRAGS], test_in[MAX_FRAGS];
-	int i;
-
-	memset(test_in, 0, sizeof(test_in));
-
-	CU_ASSERT(MAX_FRAGS >= 4);
```
Comment (anoobj) [inline test/validation/api/ipsec/ipsec_test_in.c:1870]: @JannePeltonen I remember adding this ASSERT on your insistence. I'm not able to recall why that ASSERT was required. Can you check if that is required. We can put in test_multi_out_in() if required.

File: test/validation/api/ipsec/ipsec_test_in.c [hunk 3]
```diff
@@ -1764,132 +1767,109 @@ static void test_ipsec_sa_print(void)
...
+		part_prep_esp(&test_out, 1, tunnel_ip_ver == ODPH_IPV6);
+		test_out.pkt_in = input_packets[i];
+		CU_ASSERT_EQUAL(ipsec_check_out(&test_out, out_sa, &pkt), 1);
+
+		/*
+		 * Perform inbound IPsec processing for the IPsec packet.
+		 * Expect result packet only for the last packet.
+		 */
+		memset(&test_in, 0, sizeof(test_in));
+		if (i == num_input_packets - 1) {
```
Comment (anoobj) [inline test/validation/api/ipsec/ipsec_test_in.c:1800]: 'num_input_packets' need to be atleast 1. Would this need an ASSERT?

Comment (anoobj) [reply test/validation/api/ipsec/ipsec_test_in.c:1800]: Never mind. Forgot that this was in the loop.

File: test/validation/api/ipsec/ipsec_test_in.c (hunk 2 above)
Comment (JannePeltonen) [reply test/validation/api/ipsec/ipsec_test_in.c:1870]: The assert was there to check that the array defined in the function was large enough. IOW, the code in the function assumed that MAX_FRAGS was at least 4. The new common function does not assume that and therefore does not have the assert.

Now that I look at the code that was not changed, the MAX_FRAGS stuff and the assumptions around the pkto array sizes are a bit fragile and could use a few asserts here and there. But that is for some other patch. Maybe I will take a better look and submit a new PR about it.

**Comment by JannePeltonen** [comment]
v6: Added Anoob's review tag.

**Comment by JannePeltonen** [comment]
v7: rebased

**Pull request merged**